app/views/lookup_keys/_fields.html.erb

```diff
@@ -1,60 +1,61 @@
-      <%= documentation_button('4.2.6SmartMatchers') %>
-    </div>
+    <%= param_type_selector(f, :onchange => 'keyTypeChange(this)') %>
+    <%= textarea_f f, :default_value, :value => f.object.default_value_before_type_cast, :size => "col-md-8",
+                   :disabled => (f.object.is_param && (!f.object.override || f.object.use_puppet_default)),
+                   :fullscreen => :true,
+                   :rows => :auto, :help_inline => popover('', _("Value to use when there is no match")) %>
+    <%= checkbox_f(f, :use_puppet_default, :label => _('Use Puppet default'), :size => "col-md-8",
+                   :help_inline => use_puppet_default_help,
+                   :onchange => 'toggleUsePuppetDefaultValue(this, "default_value")',
+                   :disabled => (f.object.is_param && !f.object.override)) if is_param %>
+    </br>
+  </fieldset>
+  <div class="matcher-parent" <%= "id=#{(f.object.key || 'new_lookup_keys').to_s.gsub(' ', '_')}_lookup_key_override_value" %> style=<%= "display:none;" if (f.object.is_param && !f.object.override) %>>
+    <fieldset>
+      <legend><%= _("Prioritize Attribute Order") %></legend>
+      <h6><%= popover("", _("The order in which matchers keys are processed, first match wins.<br> You may use multiple attributes as a matcher key, for example, an order of <code>host group, environment</code> would expect a matcher such as <code>hostgroup = \"web servers\", environment = production</code>"),
+                      :title => _("The order in which values are resolved")).html_safe %> <%= _("Set the order in which values are resolved") %></h6>
+      <%= textarea_f f, :path, :rows => :auto, :label => _("Order"), :id => 'order', :size => "col-md-8", :onchange => 'fill_in_matchers()', :value => f.object.path %>
+    </fieldset>
+    <%= table_checkbox_f(f, :merge_overrides, :onchange => 'mergeOverridesChanged(this)',
+                         :disabled => !f.object.supports_merge?, :size => "col-md-1", :label_size => "col-md-3",
+                         :help_inline => popover("", _("Continue to look for matches after first find (only array/hash type)? Note: merging overrides ignores all matchers that use Puppet default"))) %>
+    <%= table_checkbox_f(f, :avoid_duplicates, :disabled => (!f.object.supports_uniq? || !f.object.merge_overrides), :size => "col-md-1", :label_size => "col-md-4",
+                         :help_inline                    => popover("", _("Avoid duplicate values when merging them (only array type)?"))) %>
+    <fieldset>
+      <legend><%= _("Specify Attribute Matcher") %>  <%= documentation_button('4.2.6SmartMatchers') %></legend>
+      <div class="children_fields lookup_values">
+        <%= render 'lookup_keys/values', :f => f, :is_param => is_param %>
+      </div>
+    </fieldset>
+    <fieldset>
+      <legend><%= _("Optional input validator") %></legend>
```

Comment:
Seems like this is something you change less often than the matchers.
This was an attempt to fit the most important information to one screen without scrolling.